src/app/utils/selection.ts

```diff
@@ -45,26 +48,32 @@ function extractLinks(rawHtml: string): string[] {
     // [\n\r\s] - any whitespace character, including linebreaks
     // *? - matches any amount, lazily
     // (?!#) - makes sure that # anchors are not matched - it's unlikely that these would be desired
-    const regexp = /<[\n\r\s]*?a[\n\r\s]*?href[\n\r\s]*?=[\n\r\s]*?"[\n\r\s]*?(?<url>(?!#).*?)[\n\r\s]*?"[\S\s]*?>[\S\s]*?<[\n\r\s]*?\/[\n\r\s]*?a[\n\r\s]*?>/g;
-    
+    // const regexp = /<[\n\r\s]*?a[\n\r\s]*?href[\n\r\s]*?=[\n\r\s]*?"[\n\r\s]*?(?<url>(?!#).*?)[\n\r\s]*?"[\S\s]*?>[\S\s]*?<[\n\r\s]*?\/[\n\r\s]*?a[\n\r\s]*?>/g;
+
+    const regexp = /<a\s+(?:[^>]*?[\s"']+)?href=(["'])(?<url>.*?)\1/gim;
+
     return Array.from(rawHtml.matchAll(regexp)).map(n => n.groups.url);
 }
 
 /**
- * Adds the current page origin to the links if they are 
+ * Adds the current page origin to the links if they are
  * relative
  * @param links Input links
  */
 function addHostIfMissing(links: string[]) : string[] {
     const origin = window.location.origin;
-    
+
     return links.map(link => {
         if (link.startsWith("/")) {
             return `${origin}${link}`;
-        } else if (link.startsWith(".")) {
+        }
+        else if (link.startsWith(".")) {
             return `${origin}/${link}`;
         }
-        
+        else if (link.indexOf("mailto:") === 0) {
```

Comment:
Good idea to reject `mailto`. However, I think this function is not the right place to add such a guard. I think the regex itself could handle that:

```ts
/<a\s+(?:[^>]*?[\s"']+)?href=(["'])(?!(?:mailto:))(?<url>.*?)\1/gim
```

Reply:
Moved to regex